lib/resque/scheduler/delaying_extensions.rb

```diff
@@ -69,11 +69,25 @@ def delayed_queue_schedule_size
       end
 
       def next_delayed_items(before:, count: 1)
-        items = redis.zrangebyscore(:delayed_queue, 0.0, before.to_i, limit: [0, count], with_scores: false)
-        redis.zrem(:delayed_queue, items) unless items.empty?
+        items = evalsha(:zpop, [:delayed_queue], [0, before.to_i, 0, count])
         items.map { |item| decode_without_nonce(item) }
       end
 
+      def zpop_sha(refresh = false)
+        @acquire_sha = nil if refresh
```

Comment:
`zpop_sha`?

Reply:
yea, the method needs to be called `x_sha`. Since this effectively is a `zpop` command (get the first N items and remove them from the set), I felt it was appropriate.